packages/codeeditor/src/editor.ts

```diff
@@ -371,12 +297,17 @@ export namespace CodeEditor {
       if (this._isDisposed) {
         return;
       }
+      if (this._datastore) {
+        this._datastore.dispose();
+        this._datastore = null;
+      }
       this._isDisposed = true;
-      this.datastore.dispose();
+      this.record.datastore.dispose();
```

Comment:
Wouldn't the record need to be nulled as well (or the records ref to the datastore)?

Reply:
This is actually a mistake, I think. The intention is that if a record is passed in, then it's datastore doesn't belong to the editor, and it shouldn't dispose of it.

Reply:
Should it then not dispose of `this._datastore` either (just null it)?

Reply:
Here, `this._datastore` is only set if the editor creates it in the first place. If it is passed in to the user, it remains `null`.